### PR TITLE
Update Endpoint and add new tunnel contract

### PR DESCRIPTION
**Pull request by jfullerton44**
Fixes #

### Changes proposed: 
- Adds Id property to Endpoints
- Adds TunnelV2 which will be used in the service for the new api-version. In the future we will delete the current tunnel and rename TunnelV2 to Tunnel
-

### Other Tasks:
- [x] If you updated the Go SDK did you update the PackageVersion in tunnels.go

### CODE REVIEW

File: go/tunnels/tunnel_v.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// Generated from ../../../cs/src/Contracts/TunnelV2.cs
+
+package tunnels
+
+import (
+	"time"
+)
+
+// Data contract for tunnel objects managed through the tunnel service REST API.
+type TunnelV struct {
```
Comment (jasongin) [inline go/tunnels/tunnel_v.go:12]: Delete `tunnel_v.go`, `tunnel_v.rs`, `TunnelV.java`, `tunnelV.ts`. These files must have been unintentionally generated while you were working on the C# code, but would not be generated again.

Comment (jfullerton44) [reply go/tunnels/tunnel_v.go]: Removed

File: cs/src/Contracts/Tunnel.cs [hunk 2]
```diff
@@ -37,9 +37,9 @@ public Tunnel()
     /// Gets or sets the generated ID of the tunnel, unique within the cluster.
     /// </summary>
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    [RegularExpression(TunnelIdPattern)]
-    [StringLength(TunnelIdLength, MinimumLength = TunnelIdLength)]
-    public string? TunnelId { get; set; }
+    [RegularExpression(OldTunnelIdPattern)]
+    [StringLength(OldTunnelIdLength, MinimumLength = OldTunnelIdLength)]
+    public virtual string? TunnelId { get; set; }
```
Comment (jasongin) [inline cs/src/Contracts/Tunnel.cs:42]: Is a reason this is made `virtual`?

**Review by jasongin — APPROVED** [review]

File: cs/src/Contracts/Tunnel.cs (hunk 2 above)
Comment (jfullerton44) [reply cs/src/Contracts/Tunnel.cs]: Originally when we were going to share objects it had to be but it is not needed anymore so I removed it

**Review by jasongin — APPROVED** [review]

**Pull request merged**